source/crud/read-operations/cursor.txt

```diff
@@ -0,0 +1,174 @@
+=========================
+Access Data From a Cursor
+=========================
+
+.. default-domain:: mongodb
+
+Read operations that return multiple documents do not immediately return
+all values matching the query. Because a query can potentially match
+very large sets of documents, these operations rely upon an
+object called a :node-docs:`cursor <api/Cursor.html>`
+that fetches documents in batches to reduce both memory consumption and
+network bandwidth usage. Cursors are highly configurable and offer
+multiple interaction paradigms for different use cases.
+
+The following functions directly return cursors:
+
+- ``Collection.find()``
+
+- ``Collection.aggregate()``
+
+- ``Collection.listIndexes()``
+
+- ``Db.aggregate()``
+
+- ``Db.listCollections()``
+
+Other methods, like ``Collection.findOne()`` and ``Collection.watch()``,
+use cursors to generate results but do not directly expose those
+cursors to the method caller.
+
+Cursor Paradigms
+----------------
+
+You can work with cursors using a number of **cursor paradigms**.
+Many use cases only require one document at a time, so many of these
+cursor paradigms take advantage of this by abstracting away network
+accesses and caching behind individual document accesses. Some use cases
+require all documents at once or only a count of documents, so paradigms
+exist for those needs as well.
+
+.. warning::
+
+   Do not combine different cursor paradigms on a single cursor.
+   Operations such as ``hasNext()``, ``forEach()``, and ``toArray()``
+   each predictably modify the original cursor. If you mix these calls
+   on a single cursor, you may receive unexpected results.
+
+.. warning::
+
+   Because asynchronous calls directly modify the cursor, executing
+   asynchronous calls on a single cursor simultaneously can also cause
+   undefined behaviour. As a result, always wait for the previous
```

Comment:
Would this retain its original intent by omitting "As a result" so that the imperative "always wait..." is more immediate? Also, would use "one" in the second mention of "asynchronous operation".

Reply:
Getting rid of "as a result", but I'm not 100% sold on changing that second mention of "asynchronous operation" to "one". Instead, I'm just ending with "another" -- I think that solves things nicely without resorting to a vague pronoun.